Library/Homebrew/formula_installer.rb

```diff
@@ -31,16 +31,36 @@ class FormulaInstaller
   include FormulaCellarChecks
   extend Attrable
 
-  attr_reader :formula, :bottle_tab_runtime_dependencies
+  ETC_VAR_DIRS = T.let([HOMEBREW_PREFIX/"etc", HOMEBREW_PREFIX/"var"].freeze, T::Array[Pathname])
 
-  attr_accessor :options, :link_keg
+  sig { override.returns(Formula) }
+  attr_reader :formula
+
+  sig { returns(T::Hash[String, T::Hash[String, String]]) }
+  attr_reader :bottle_tab_runtime_dependencies
+
+  sig { returns(Options) }
+  attr_accessor :options
+
+  sig { returns(T::Boolean) }
+  attr_accessor :link_keg
 
   attr_predicate :installed_as_dependency?, :installed_on_request?
   attr_predicate :show_summary_heading?, :show_header?
   attr_predicate :force_bottle?, :ignore_deps?, :only_deps?, :interactive?, :git?, :force?, :overwrite?, :keep_tmp?
   attr_predicate :debug_symbols?
   attr_predicate :verbose?, :debug?, :quiet?
 
+  sig {
+    params(formula: Formula, link_keg: T::Boolean, installed_as_dependency: T::Boolean,
+           installed_on_request: T::Boolean, show_header: T::Boolean, build_bottle: T::Boolean,
+           skip_post_install: T::Boolean, force_bottle: T::Boolean, bottle_arch: T.nilable(String),
+           ignore_deps: T::Boolean, only_deps: T::Boolean, include_test_formulae: T::Array[Formula],
+           build_from_source_formulae: T::Array[Formula], env: T.nilable(String), git: T::Boolean,
+           interactive: T::Boolean, keep_tmp: T::Boolean, debug_symbols: T::Boolean, cc: T.nilable(String),
+           options: Options, force: T::Boolean, overwrite: T::Boolean, debug: T::Boolean, quiet: T::Boolean,
+           verbose: T::Boolean).void
```

Comment:
This is kind of unreadable. I understand it will be gigantic, but I suggest breaking this up so we have one `param` per line.

Reply:
Done! (I might also suggest ordering the optional kwargs, but that feels like scope creep here…)